code/game/objects/items/implants/implant_misc.dm

```diff
@@ -173,3 +173,13 @@
 /obj/item/implanter/radio/syndicate/selfdestruct
 	name = "implanter (modified internal syndicate radio)"
 	imp_type = /obj/item/implant/radio/syndicate/selfdestruct
+
+/obj/item/implant/undercover
+	name = "undercover implant"
```

Comment:
What’s wrong with a normal mindshield? Pretty snowflakey item

Reply:
> What’s wrong with a normal mindshield? Pretty snowflakey item

The average sec player is too stupid to notice that an undercover cop is mindshielded.

Reply:
Basically that.

They may freak when mindshielded assistant opens brig doors.

Reply:
I think that's better since it adds more options for people who get their hands on mindshield to say something like 'im detective'

Reply:
> > What’s wrong with a normal mindshield? Pretty snowflakey item
> 
> The average sec player is too stupid to notice that an undercover cop is mindshielded.

If the Detective is running an undercover gimmick and gets temporarily arrested because they didn't inform anyone they were running it, _good_.